### PR TITLE
Create EndPointConfig object from model_server in /model

### LINKED ISSUE #3770: PUT /model with model_server fails: 'dict' object has no attribute 'url'
**Rasa version**: Rasa 1.1.0a1

**Python version**: Python 3.6.8

**Operating system** (windows, osx, ...): Docker image rasa/rasa:latest-full
Linux 6f5ab25d1c4d 4.15.0-47-generic #50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019 x86_64 GNU/Linux

**Issue**: using model_server endpoint for https://rasa.com/docs/rasa/api/http-api/#tag/Model/paths/~1model/put leads to error 
AttributeError: 'dict' object has no attribute 'url'

**Error (including full traceback)**:
```
  File "/usr/local/lib/python3.6/site-packages/rasa/server.py", line 258, in _load_agent
    action_endpoint=action_endpoint,
  File "/usr/local/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/agent.py", line 246, in load_agent
    model_server,
  File "/usr/local/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/agent.py", line 53, in load_from_server
    await _update_model_from_server(model_server, agent)
  File "/usr/local/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/usr/local/lib/python3.6/site-packages/rasa/core/agent.py", line 107, in _update_model_from_server
    if not is_url(model_server.url):
AttributeError: 'dict' object has no attribute 'url'
```

**Command or request that led to error**:
```
curl -X PUT -H 'Content-Type: application/json' -i http://localhost:5005/model --data '{"model_file": null, "model_server": {"url": "http://localhost:5000/model/8b200dad622c4ed9b9636e8ec7b12660", "params": {}, "headers": {}, "basic_auth": {}, "token": "", "token_name": "", "wait_time_between_pulls": 0}}'
```

Issue seems to be because of following files:

https://github.com/RasaHQ/rasa/blob/e4d3335a7834b56ab52c68bb30ae58b94e9b6ddd/rasa/server.py#L818

model_server is just a get on json, never converted to an object.

https://github.com/RasaHQ/rasa/blob/e4d3335a7834b56ab52c68bb30ae58b94e9b6ddd/rasa/core/agent.py#L107

Accessing the json object (dict) as if its an EndpointConfig

This pull request fixes it: https://github.com/RasaHQ/rasa/pull/3769

### PR DESCRIPTION
**Pull request by mmalhotra**
**Proposed changes**:
- Fixed /model put with model_server
- fixes #3770

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible!

File: rasa/server.py [hunk 1]
```diff
@@ -817,7 +817,17 @@ def validate_request(rjs):
...
-
+        if model_server:
+            try:
+                model_server = EndpointConfig.from_dict(model_server)
+            except TypeError as e:
+                logger.debug(traceback.format_exc())
+                raise ErrorResponse(
+                    400,
+                    "BadRequest",
+                    "Supplied model_server is not valid. {}".format(e),
```
Comment (tabergma) [inline rasa/server.py]:
```suggestion
                    "Supplied 'model_server' is not valid. Error: {}".format(e),
```

**Comment by mmalhotra** [comment]
@tabergma Thanks, always happy to help this wonderful project!
I'm having a hard time figuring out why the tests are failing even though I haven't touched the part of code that could make those tests fail. Any tips?

**Comment by tabergma** [comment]
Looks like you found the issue 🚀 I guess once the conflicts are resolved, we can merge.

**Comment by tabergma** [comment]
FYI Looking into the failed tests.

File: tests/test_server.py [hunk 2]
```diff
@@ -661,6 +663,38 @@ def test_load_model(rasa_app, trained_core_model):
...
+            assert response.status == 204
+
+            _, response = rasa_app.get("/status")
+
+            assert response.status == 200
+            assert "fingerprint" in response.json
+
+            assert old_fingerprint != response.json["fingerprint"]
+
+
```
Comment (tabergma) [inline tests/test_server.py]:
```suggestion
    import rasa.core.jobs
    rasa.core.jobs.__scheduler = None
```

File: tests/test_server.py [hunk 3]
```diff
@@ -661,6 +663,39 @@ def test_load_model(rasa_app, trained_core_model):
...
+            _, response = rasa_app.get("/status")
+
+            assert response.status == 200
+            assert "fingerprint" in response.json
+
+            assert old_fingerprint != response.json["fingerprint"]
+
+    import rasa.core.jobs
+    rasa.core.jobs.__scheduler = None
```
Comment (tabergma) [inline tests/test_server.py]:
```suggestion

    rasa.core.jobs.__scheduler = None


```

Comment (tabergma) [reply tests/test_server.py]: Black is failing. The file needs to be formatted.

**Comment by tabergma** [comment]
@mmalhotra I have no idea why the test `test_warning_for_base_paths_with_trailing_slash` is failing. As the test is not that important, can you please just remove it? Hopefully, we can finally merge the PR afterwards. Thanks.

**Comment by mmalhotra** [comment]
@tabergma  the check succeeded at https://travis-ci.com/RasaHQ/rasa/builds/118347721
it still shows "Expected — Waiting for status to be reported" here, rerun maybe?

**Comment by tabergma** [comment]
Thanks for fixing the test!

**Pull request merged**